### ISSUE TITLE
Backup to S3 won't allow bucket names with special characters

### ISSUE THREAD

**Issue by kuking** [mail-in-a-box/mailinabox#565]
While configuring the backup system to upload a copy to Amazon S3; if the bucket name has dots or dashes in it, it will fail with the following error message:

```hostname 'mail.xxx.tld-bk.s3-sa-east-1.amazonaws.com' doesn't match either of 's3-sa-east-1.amazonaws.com', '*.s3-sa-east-1.amazonaws.com'```

* Different alternatives have be entried for the s3 path, i.e. ```bucket/path```,  ```s3://bucket/path``` and ```//bucket/path```
* The bucket name mail.xxx.tld-bk has been anonymised.

Thanks for the fantastic project.

**Comment by JoshData** [comment]
May be https://github.com/boto/boto/issues/2836 or https://github.com/boto/boto/issues/443. Not sure what the right thing to do here is.

**Comment by kuking** [comment]
yep looks similar. I guess wait until upstream gets fixed. Nevertheless, this seems to be a workaround: https://github.com/boto/boto/issues/2836#issuecomment-77283169

**Comment by elsiehupp** [comment]
[This is still a problem.](https://discourse.mailinabox.email/t/s3-not-connecting-on-two-servers/4918/4)

At the very least it would be nice if the admin interface gave a more accurate error message, rather than making the incorrect statement that hostname doesn't match.

**Comment by lpossamai** [comment]
+1 - I'm also having this issue
